library/general/src/lib/config_file/config_file.rb

```diff
@@ -0,0 +1,29 @@
+module ConfigFile
+  class ConfigFile
+    def initialize(parser, file_path, file_class: File)
+      @file_class = file_class
+      @parser = parser
+      @file_path = file_path
+    end
+
+    def write(changes_only: false)
```

Comment:
whatever agent needs. In general common base support read/write, no caching, only parsed tree is stored in memory, so you can call more writes...if you need you can call more reads to reset.